### PR TITLE
Made WPCS compliant. Added tests.

### PR DESCRIPTION
**Pull request by hellofromtonya**
Functions: `beans_remove_action` and `beans_reset_action`.

1. Made WPCS compliant.
2. Added integration tests.
3. Added unit tests.
4. Improved documentation.
5. Improved both functions.

### CODE REVIEW

**Comment by hellofromtonya** [comment]
@ThierryA @christophherr This PR makes both the `beans_remove_action` and `beans_reset_action` compliant and fully tested.

@christophherr It includes the changes you noted from PR #102.

Upon merging this one, I'll then go back and update all of the other Actions functions to use our new test data fixture.

File: tests/phpunit/integration/api/actions/beansRemoveAction.php [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+			// Check that the action is registered with Beans.
+			$this->assertSame( $action, _beans_get_action( $beans_id, 'added' ) );
+
+			// Check that the action is registered with WordPress.
+			$this->assertTrue( has_action( $action['hook'], $action['callback'] ) !== false );
+
+			// Do the remove.
+			$this->assertSame( $action, beans_remove_action( $beans_id ) );
+
+			// Check what is stored in "removed".
```
Comment (christophherr) [inline tests/phpunit/integration/api/actions/beansRemoveAction.php]: Should this be similar to L#72?
`Check if the action is stored as "removed".`

File: tests/phpunit/integration/api/actions/beansRemoveAction.php [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+			$this->assertSame( $action, _beans_get_action( $beans_id, 'removed' ) );
+
+			// Check that is no longer registered with WordPress, i.e. that `remove_action` did run.
```
Comment (christophherr) [inline tests/phpunit/integration/api/actions/beansRemoveAction.php]: `Check that the action is no longer...`

File: tests/phpunit/unit/api/actions/beansResetAction.php [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+	 */
+	public function test_should_reset_after_modifying_the_callback() {
+		$modified_action = array(
+			'callback' => 'my_callback',
+		);
+
+		$this->go_to_post();
+
+		foreach ( static::$test_actions as $beans_id => $action ) {
+			// Test the original callback.
```
Comment (christophherr) [inline tests/phpunit/unit/api/actions/beansResetAction.php]: Should we stay consistent with the previous comments and say something like
`Check that the original callback is registered before we start.`?

File: tests/phpunit/unit/api/actions/beansResetAction.php [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+	 *
+	 * Why? "Replace" overwrites and is not resettable.
+	 */
+	public function test_should_not_reset_after_replacing_hook() {
+		$hook = 'foo';
+
+		$this->go_to_post();
+
+		foreach ( static::$test_actions as $beans_id => $action ) {
+			// Test the action before we start.
```
Comment (christophherr) [inline tests/phpunit/unit/api/actions/beansResetAction.php]: Similar to L#106 above.

File: tests/phpunit/unit/api/actions/beansResetAction.php [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+	 *
+	 * Why? "Replace" overwrites and is not resettable.
+	 */
+	public function test_should_not_reset_after_replacing_callback() {
+		$callback = 'foo_cb';
+
+		$this->go_to_post();
+
+		foreach ( static::$test_actions as $beans_id => $action ) {
+			// Test the action before we start.
```
Comment (christophherr) [inline tests/phpunit/unit/api/actions/beansResetAction.php]: Similar to L#106 and L#135 above.

File: tests/phpunit/integration/api/actions/beansRemoveAction.php (hunk 1 above)
Comment (christophherr) [reply tests/phpunit/integration/api/actions/beansRemoveAction.php]: Or similar to  the previous PR: 
`Check that it was stored as "removed"

File: tests/phpunit/unit/api/actions/beansResetAction.php [hunk 6]
```diff
@@ -100,24 +94,231 @@ public function test_should_reset_after_modifying_the_callback() {
...
+			Monkey\Actions\expectAdded( $action['hook'] )
+				->once()
+				->whenHappen( function( $callback, $priority, $args ) use ( $action, $modified_action ) {
+					// Check that all of the parameters were modified in WordPress.
+					$this->assertSame( $modified_action['callback'], $callback );
+					$this->assertSame( $modified_action['priority'], $priority );
+					$this->assertSame( $modified_action['args'], $args );
+				} );
+
+			// Modify the action's callback and priority.
```
Comment (christophherr) [inline tests/phpunit/unit/api/actions/beansResetAction.php]: `callback, priority, and args.`

**Pull request merged**